models/issues/issue_dev_link.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package issues
+
+import (
+	"context"
+	"strconv"
+
+	"code.gitea.io/gitea/models/db"
+	git_model "code.gitea.io/gitea/models/git"
+	repo_model "code.gitea.io/gitea/models/repo"
+	"code.gitea.io/gitea/modules/timeutil"
+)
+
+type IssueDevLinkType int
+
+const (
+	IssueDevLinkTypeBranch IssueDevLinkType = iota + 1
```

Comment:
Don't understand why it should link to a branch without PR.

`Create a branch from an issue, with pull/merge request #20226` said that "a branch with PR"